indexer/config/config.go

```diff
@@ -34,15 +36,38 @@ type L1Contracts struct {
 	// Remove afterwards?
 }
 
+// converts struct of to a slice of addresses for easy iteration
+// also validates that all fields are addresses
+func (c *L1Contracts) AsSlice() ([]common.Address, error) {
```

Comment:
Rather than `AsSlice` how about a `ForEach` that takes a func? That way we can retain the logs on startup that spits out the configured (name, contract_address) pairs?